src/persist-types/src/schema.rs

```diff
@@ -199,7 +213,14 @@ fn backward_compatible_typ(old: &DataType, new: &DataType) -> Option<ArrayMigrat
         (FixedSizeBinary(_), _) => None,
         (Struct(o), Struct(n)) => backward_compatible_struct(o, n),
         (Struct(_), _) => None,
-        (List(o), List(n)) => (o == n).then_some(NoOp),
+        (List(o), List(n)) | (Map(o, _), List(n)) => {
```

Comment:
I'd like for it to be an invariant that two equal datatypes produce `Some(NoOp)` (and not just a migration whose logic happens to be a noop). The proptests aren't testing this at the moment because `NoOp` isn't public to mz-storage-types, but maybe it's time to make that happen.